test/refactoring/ConvertAsync/basic.swift

```diff
@@ -656,3 +656,209 @@ func testCalls() {
   // VOID-AND-ERROR-CALL4-NEXT: {{^}}print("void and error completion \(<#v#>)"){{$}}
 }
 // CONVERT-FUNC: {{^}}}
+
+func testSimpleWithVariableCompletionHandler(completionHandler: (String) -> Void) {
+  // RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=SIMPLE-WITH-VARIABLE-COMPLETION-HANDLER %s
+  simple(completion: completionHandler)
+  // SIMPLE-WITH-VARIABLE-COMPLETION-HANDLER: let result = await simple()
+  // SIMPLE-WITH-VARIABLE-COMPLETION-HANDLER: completionHandler(result)
+}
```

Comment:
Also, if these *were* treated as completion handler functions, what happens when converting/adding async? eg.
```
func simple(_ completionHandler: () -> Void) {}

// Convert/add on `delay`
func delay(_ function: () -> Void) {
  simple(function)
}
```

Reply:
> IMO I'd name completionHandler something else for these tests, since otherwise they look like completion handler functions, but we're using convert call instead of convert function. maybe runAfter or something?

`testSimpleWithVariableCompletionHandler` is essentially just a wrapper around `simple`. The user might want to do some logging here, convert arguments etc, so I think `completionHandler` is the correct name. I 
deliberately didn’t use `completion` to make sure we are not picking up the variable name from the `simple` function declaration.

> The function as-is looks like it should just be converted to

I don’t think I follow. Is this the same point as the one below?

> Also, if these were treated as completion handler functions, what happens when converting/adding async? eg.

Good point. With this change, we rewrite
```swift
// RUN: %refactor -convert-to-async -dump-rewritten -source-filename %s -pos=%(line+1):1
func testSimpleWithVariableCompletionHandler(completionHandler: (String) -> Void) {
  simple(completion: completionHandler)
}
```

to 

```swift
func testSimpleWithVariableCompletionHandler() async -> String {
  let result = await simple()
  completionHandler(result)
}
```

which is definitely not correct.